lib/common/http2client.c

```diff
@@ -282,6 +283,26 @@ static void call_stream_callbacks_with_error(struct st_h2o_http2client_conn_t *c
     kh_foreach_value(conn->streams, stream, { call_callback_with_error(stream, errstr); });
 }
 
+static int extract_content_length(const h2o_headers_t *headers, size_t *content_length, const char **err_desc)
+{
+    *content_length = SIZE_MAX;
+    for (size_t i = 0; i < headers->size; ++i) {
+        if ((const h2o_token_t *)headers->entries[i].name == H2O_TOKEN_CONTENT_LENGTH) {
+            const h2o_iovec_t *value = &headers->entries[i].value;
+            if (*content_length != SIZE_MAX) {
+                *err_desc = "duplicate content-length";
+                return -1;
+            }
+            *content_length = h2o_strtosize(value->base, value->len);
+            if (*content_length == SIZE_MAX) {
+                *err_desc = "malformed content-length";
+                return -1;
+            }
+        }
+    }
+    return 0;
```

Comment:
I think it'd be more consistent with other parts of the code to return a boolean indicating success (i.e. 1 on success, 0 on failure), than using 0 for success and -1 for failure.

Reply:
Done in bc4b2c858.